weather_api.go

```diff
@@ -42,10 +60,6 @@ func (c WeatherAPIConfig) GetCurrentWeather(options Options) (*models.WeatherRes
 
 // makeRequest is a helper function for making API requests.
 func (c WeatherAPIConfig) makeRequest(method, url string) (*http.Response, error) {
-	client := &http.Client{
-		Timeout: time.Second * config.APIsTimeout,
-	}
-
 	req, err := http.NewRequest(method, url, http.NoBody)
```

Comment:
No body is fine if we are only doing GET here

Reply:
As of now there is only GET requests, In future if we do any other request, I'll modify this.